lib/apollo-federation/entities_field.rb

```diff
@@ -49,6 +49,14 @@ def _entities(representations:)
         # TODO: What if the type is an interface?
         type_class = class_of_type(type)
 
+        if type_class.underscore_reference_keys
+          references.map! do |reference|
```

Comment:
Does this not still mutate? Could we do something like `references = references.map` here to avoid changing anything about the original references array or its members?

Reply:
This is mutating the local `references` array allocated a few lines up on [this](https://github.com/Gusto/apollo-federation-ruby/pull/248/files#diff-353631238a3878fef788c3ab953b5d493aabdb176b7a23b52deda2b305955e6fL38) line. The original `representations` method argument and its members aren’t modified with these latest changes.

Reply:
Oops. Missed that. Thanks for pointing it out.